og_context/og_context.module

```diff
@@ -200,24 +200,43 @@ function og_context($group_type = 'node', $group = NULL, $account = NULL, $check
   // User account is sent.
   $account = $account ? $account : user_load($user->uid);
 
-  $context = &drupal_static(__FUNCTION__, FALSE);
+  $id = FALSE;
+  if ($group) {
+    list($id) = entity_extract_ids($group_type, $group);
+  }
+
+  $cache_keys = array(
+    $group_type,
+    $id,
+    $account->uid,
+    $check_access,
+  );
+  $cache_key = implode(':', $cache_keys);
+
+  $context = &drupal_static(__FUNCTION__, array());
+
+  if (empty($context[$cache_key])) {
```

Comment:
this looks wrong - if we called ``og_context()`` and it was empty, then it will enter here again in the second call.